include/xtensor/xchunked_array.hpp

```diff
@@ -5,69 +5,94 @@
 namespace xt
 {
     template <class chunk_type>
-    class xchunked_array
+    class xchunked_array: public xt::xaccessible<xchunked_array<chunk_type>>
     {
     public:
 
         using const_reference = typename chunk_type::const_reference;
+        using reference = typename chunk_type::reference;
 
         template <class... Idxs>
         inline const_reference operator()(Idxs... idxs) const
         {
-            auto chunk_indexes_packed = get_chunk_indexes(std::make_index_sequence<sizeof...(Idxs)>(), idxs...);
-            auto chunk_indexes = unpack(chunk_indexes_packed);
-            auto indexes_of_chunk(std::get<0>(chunk_indexes));
-            auto indexes_in_chunk(std::get<1>(chunk_indexes));
-            chunk_type chunk = m_chunks.element(indexes_of_chunk.cbegin(), indexes_of_chunk.cend());
-            const_reference val = chunk.element(indexes_in_chunk.cbegin(), indexes_in_chunk.cend());
-            int di = 0;
-            for (auto index_of_chunk: indexes_of_chunk)
-            {
-                auto index_in_chunk = indexes_in_chunk[di];
-                di++;
-            }
-            return val;
+            auto ii = get_indexes(idxs...);
+            auto& chunk = m_chunks.element(ii.first.cbegin(), ii.first.cend());
+            return chunk.element(ii.second.cbegin(), ii.second.cend());
         }
 
+        template <class... Idxs>
+        inline reference operator()(Idxs... idxs)
+        {
+            auto ii = get_indexes(idxs...);
+            auto& chunk = m_chunks.element(ii.first.cbegin(), ii.first.cend());
+            return chunk.element(ii.second.cbegin(), ii.second.cend());
+        }
 
-        xchunked_array(std::vector<size_t> shape, std::vector<size_t> chunks):
+        xchunked_array(std::vector<size_t> shape, std::vector<size_t> chunk_shape):
```

Comment:
This can be replaced with a template constructor (whose template parameter is the type of `shape` and `chunk_shape`).